_posts/2020-09-17-accessing-snowplow-data-in-real-time-aws/index.md

```diff
@@ -0,0 +1,123 @@
+        try:
+            # Transform record from Snowplow TSV to JSON
+            print(snowplow_analytics_sdk.event_transformer.transform(record))
+            print("Succesfully retrieved and transformed event")
+        except snowplow_analytics_sdk.snowplow_event_transformation_exception.SnowplowEventTransformationException as e:
+            for error_message in e.error_messages:
+                print(error_message)
+            continue
+        
+    return
+```
+
+## 5. Package and upload the Python script
+Package a zip file for lambda with the following script. Copy it to a new file and place it in the directory above the python script. Note that this is being run on a subdirectry with the Python app called lambda_function_payload.py. The Python fuction above is written in Python 3 the script below will install the [Snowplow Python Analytics SDK](https://github.com/snowplow/snowplow/wiki/Python-Analytics-SDK-Setup) in the directory so that it is included in the zip file.
+
+```bash
+#!/bin/bash
+cd lambda_function_payload
+pip3 install snowplow_analytics_sdk -t . --upgrade
+chmod -R 755 .
+zip -r ../lambda_function_payload.zip .
+cd ..
+```
+
+Upload the zip file to lambda in the AWS Console. Since the dependency is a couple of mb, we won't be able to see and edit the code in AWS. 
+![Screenshot](img/upload_zip.png)
+
+Set up the basic settings to point to the handler in the Python script. The handler is the entry point into the code where execution will begin. The handler is formated as`name_of_python_file.name_of_handler_function`. The `.py` is not included.
+![Screenshot](img/basic_settings.png)
+
+## 6. Add a trigger to connect it to the enriched kinesis stream
+Set up your trigger based on the screenshot below. The default values will be fine but note that your Kinesis stream will be named differently. 
+![Screenshot](img/kinesis_setup.png)
+
+## 7. Test and confirm events are being processed
+Send some events into your Snowplow and give them a couple seconds to process. Click on monitoring on the lambda fuction page and scroll down to CloudWatch logs. You should see some entries appearing. 
+![Screenshot](img/monitoring.png)
+![Screenshot](img/logs.png)
+
+Looking at the logs, you should see something like this. What you see here is the entire event's data in JSON format. You have now succesfully read from the real time stream and are ready to process your events!
+![Screenshot](img/log_details.png)
+
+# Okay, what next?
+Now that you have control over data coming out of the real time stream. Here are some articles with ideas on what you can do with it:
+*   [How real-time data enables personalization and engagement](https://snowplowanalytics.com/blog/2019/09/27/how-real-time-data-lets-media-companies-personalize-content-messaging-and-advertising/)
+*   [Snowplow for retail part 5](https://snowplowanalytics.com/blog/2019/03/06/snowplow-for-retail-part-5-what-can-we-do-with-data-when-were-well-established/)
```

Comment:
It's pretty in depth but would linking to the CodeCombat real time example on Discourse be a good whats next article too?